flowredux/src/commonMain/kotlin/com/freeletics/flowredux/sideeffects/OnEnterStartStateMachine.kt

```diff
@@ -3,99 +3,48 @@ package com.freeletics.flowredux.sideeffects
 import com.freeletics.flowredux.dsl.ChangedState
 import com.freeletics.flowredux.dsl.State
 import com.freeletics.flowredux.util.CoroutineWaiter
-import com.freeletics.flowredux.util.mapToIsInState
-import com.freeletics.flowredux.util.whileInState
 import com.freeletics.mad.statemachine.StateMachine
 import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.coroutineScope
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.emptyFlow
+import kotlinx.coroutines.flow.channelFlow
 import kotlinx.coroutines.flow.flatMapConcat
-import kotlinx.coroutines.flow.flatMapLatest
-import kotlinx.coroutines.flow.onCompletion
-import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.mapNotNull
 import kotlinx.coroutines.flow.onStart
 import kotlinx.coroutines.launch
 
 @ExperimentalCoroutinesApi
 internal class OnEnterStartStateMachine<SubStateMachineState : Any, SubStateMachineAction : Any, InputState : S, S : Any, A>(
     override val isInState: IsInState<S>,
-    private val subStateMachineFactory: (InputState) -> StateMachine<SubStateMachineState, SubStateMachineAction>,
```

Comment:
No factory needed anymore in the future or is that something that will come later?

Reply:
I wanted to keep it consistent with `onEnter` where we use the initialState from the builder. We could pass both the factory and the initialState here but there isn't really a point in that.